mypy/join.py

```diff
@@ -17,6 +17,11 @@
 def join_simple(declaration: Type, s: Type, t: Type) -> Type:
     """Return a simple least upper bound given the declared type."""
 
+    if (s.can_be_true, s.can_be_false) != (t.can_be_true, t.can_be_false):
```

Comment:
I tried to test and couldn't make a failing case with join_type; I think
that one always creates new instances and new instances have always
undetermined boolean values. The problem with join_simple is that it was
reusing the existing types

On Tue, Aug 16, 2016 at 8:21 PM, David Fisher <notifications@github.com>
wrote:

> In mypy/join.py
> <https://github.com/python/mypy/pull/1989#discussion_r75002438>:
>
> > @@ -17,6 +17,11 @@
> >  def join_simple(declaration: Type, s: Type, t: Type) -> Type:
> >      """Return a simple least upper bound given the declared type."""
> >
> > +    if (s.can_be_true, s.can_be_false) != (t.can_be_true, t.can_be_false):
>
> We probably need this for join_types also, right?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/python/mypy/pull/1989/files/6026e66043dc3827ab2a695e76d5c6d190756432#r75002438>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAMQHZnI4JKGVVRd5VePq5btvg3MW3SJks5qgg2ggaJpZM4Jd5oe>
> .
>



-- 
Daniel F. Moisset - UK Country Manager
www.machinalis.com
Skype: @dmoisset